reed-solomon-novelpoly/src/novel_poly_basis/tests.rs

```diff
@@ -463,3 +464,32 @@ fn shard_len_is_reasonable() {
 	// needs 3 bytes to fit, rounded up to next even number.
 	assert_eq!(rs.shard_len(19), 6);
 }
+
+#[derive(Clone, Debug)]
+struct ArbitraryData(Vec<u8>);
+
+impl Arbitrary for ArbitraryData {
+	fn arbitrary(g: &mut Gen) -> Self {
+		// Limit the len to 1 mib, otherwise the test will take forever
+		let len = u32::arbitrary(g).saturating_add(2) % (1024 * 1024);
```

Comment:
what is this saturating add for? if we want len at least 2, it should be added after modulus operation or simply .max(2)

Reply:
yes, to not get a zeroed length.
doesn't that achieve exactly the same thing as my code?

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aa3f3dd2f1cd8d82eea0b80c65b3a3bb

Reply:
didn't think of this 👍🏻 thanks